types/bootstrap/test/modal-tests.ts

```diff
@@ -18,23 +18,28 @@ Modal.VERSION;
 Modal.Default;
 
 element.addEventListener(Modal.Events.show, event => {
-    // do something…
+    event.target; // $ExpectType HTMLElement
+    event.relatedTarget; // $ExpectType HTMLElement | undefined
 });
 
 element.addEventListener(Modal.Events.shown, event => {
-    // do something…
+    event.target; // $ExpectType HTMLElement
+    event.relatedTarget; // $ExpectType HTMLElement | undefined
 });
 
 element.addEventListener(Modal.Events.hide, event => {
-    // do something…
+    event.target; // $ExpectType HTMLElement
+    event.relatedTarget; // $ExpectType HTMLElement | undefined
```

Comment:
Note that these changes imply that an event for `hide.bs.modal`/`hidden.bs.modal`/`hidePrevented.bs.modal` could have a `relatedTarget` property, even though in practice that could never happen. It would be possible to fix that with something like the following:

```ts
declare namespace Modal {
    interface Event extends CustomEvent {
        target: HTMLElement;
    }
    interface ShowEvent extends Event {
        relatedTarget?: HTMLElement;
    }
}

declare global {
    interface Element {
        addEventListener(
            type: Modal.Events.show | Modal.Events.shown,
            listener: (this: Element, ev: Modal.ShowEvent) => any,
            options?: boolean | AddEventListenerOptions,
        ): void;
        addEventListener(
            type: Modal.Events.hide | Modal.Events.hidden | Modal.Events.hidePrevented,
            listener: (this: Element, ev: Modal.Event) => any,
            options?: boolean | AddEventListenerOptions,
        ): void;
    }
}
```

It's not clear to me whether the increased complexity is worth it for slightly more precise types. Input from maintainers would be appreciated!

Reply:
@martin-badin Thanks for the review! Do you have any opinion about this?